src/dmd/dinterpret.d

```diff
@@ -6095,6 +6095,30 @@ public:
             se = (cast(ClassReferenceExp)ex).value;
             i = (cast(ClassReferenceExp)ex).findFieldIndexByName(v);
         }
+        else if (ex.op == TOK.typeid_)
+        {
+            auto tie = cast(TypeidExp) ex;
+            if (v.ident == Identifier.idPool("name"))
+            {
+                if (auto t = isType(tie.obj))
+                {
+                    auto sym = t.toDsymbol(null);
+                    auto ident = (sym ? sym.ident : null);
+                    if (ident)
```

Comment:
`if (auto ident = sym ? sym.ident : null)`

Reply:
I don't want to increase the nesting level unnecessarily.

Reply:
The suggested change does not affect the nesting level, it simply limits the scope of `ident` to the block where it is used, getting rid of a line of code. This is the standard policy in such situations throughout the dmd codebase.

Reply:
ah yes can do.